Is_Subtree.java

```diff
@@ -0,0 +1,74 @@
+/*1. PROBLEM STATEMENT:
+ * Given two non-empty binary trees s and t, check whether tree t has exactly the same structure and node values with a subtree of s. A subtree of s is a tree consists of a node in s and all of this node's descendants. The tree s could also be considered as a subtree of itself.
+2. EXAMPLES:
+Example 1:
+Given tree s:
+
+     1
+    / 
+ 	1
+Given tree t:
+  1
+Return true, because t has the same structure and node values with a subtree of s.
+Example 2:
+Given tree s:
+
+     3
+    / \
+   4   5
+  / \
+ 1   2
+
+Given tree t:
+   4
+  / 
+ 1   
+Return false.
+3. SOLUTION:
+Recursively check 's' tree until 't' is found as a subtree in 's'
+*/
+package Week6;
+
+import java.util.Scanner;
+
+public class Is_Subtree {
+
+	public static void main(String[] args) {
+		Scanner in = new Scanner(System.in);
+		TreeNode s = constructTree();
+		TreeNode t = constructTree();
+		Subtree_Of_Another_Tree sol = new Subtree_Of_Another_Tree();
+		System.out.println(sol.isSubtree(s, t));
+
+	}
+	public static TreeNode constructTree() {
+
+		Scanner s = new Scanner(System.in);
+		int rootData = s.nextInt();
+		if (rootData != -1) {
+			TreeNode root = new TreeNode(rootData);
+			root.left = constructTree();
+			root.right = constructTree();
+			return root;
+		}
+		return null;
+	}
+	public boolean isSubtree(TreeNode s, TreeNode t) {
+		return isSubtreeH(s,t,false);
+	}
+	private boolean isSubtreeH(TreeNode s, TreeNode t,boolean res) {

```

Comment:
What does the 'H' in isSubtreeH mean? Helper? 

It is actually ok to just name it isSubtree as that means the method is overloaded because it has different parameters anyways. If you want to name it a helper method, write out Helper instead of just H